test/caching.js

```diff
@@ -0,0 +1,146 @@
+// Copyright 2019 Google LLC
```

Comment:
nit: All the js tests are linked together on test/test.html page now and can be run all together by opening that page. They are there, because they use some DOM elements. I think it would be better if you linked this file there also, but this is not necessary.

Reply:
Added it to test.html, thank you!